### ISSUE TITLE
qTox fails to cross-compile to Windows due to -Werror

### ISSUE THREAD

**Issue by nurupo** [qTox/qTox#4707]
@Diadlo what in the world were you thinking when you added `-Werror` in a release configuration? Are you out of your mind? Do you realize that, for example, while gcc 4.9 might give you 0 warnings, gcc 7.0 might have introduce new warnings that would get triggered or clang might have an entirely different set of warnings than gcc 4.9 has? Please don't enable `-Werror` in release, it's very compiler dependent, compiler version dependent, and even platform-dependent, so  I'd even argue as far as not having it in debug, but having a separate cmake option for enabling `-Werror`.

**Just to make it clear again, it's not only an issue when trying to cross-compile to Windows, just using a different compiler version or different compiler is enough. You should release v1.12.1 with `-Werror` removed from default/release build, as some compilers won't be able to get qTox to compile because of this.**

Here is build log when trying to cross-compile to Windows using [`windows/cross-compile`](https://github.com/qTox/qTox/tree/master/windows/cross-compile):

```
+ cmake -DCMAKE_TOOLCHAIN_FILE=./toolchain.cmake -DCMAKE_BUILD_TYPE=Release ..
-- The C compiler identification is GNU 5.3.1
-- The CXX compiler identification is GNU 5.3.1
-- Check for working C compiler: /usr/bin/i686-w64-mingw32-gcc
-- Check for working C compiler: /usr/bin/i686-w64-mingw32-gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/i686-w64-mingw32-g++
-- Check for working CXX compiler: /usr/bin/i686-w64-mingw32-g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Checking for one of the modules 'libavcodec'
-- LIBAVCODEC LIBRARY_DIRS: /workspace/i686/dep-cache/libffmpeg/lib
-- LIBAVCODEC INCLUDE_DIRS: /workspace/i686/dep-cache/libffmpeg/include
-- LIBAVCODEC CFLAGS_OTHER: 
-- LIBAVCODEC LIBRARIES:    avcodec;avicap32;gdi32;psapi;ole32;strmiids;uuid;oleaut32;shlwapi;secur32;m;psapi;advapi32;shell32;avutil;m
-- LIBAVCODEC found
-- Checking for one of the modules 'libavdevice'
-- LIBAVDEVICE LIBRARY_DIRS: /workspace/i686/dep-cache/libffmpeg/lib
-- LIBAVDEVICE INCLUDE_DIRS: /workspace/i686/dep-cache/libffmpeg/include
-- LIBAVDEVICE CFLAGS_OTHER: 
-- LIBAVDEVICE LIBRARIES:    avdevice;avicap32;gdi32;psapi;ole32;strmiids;uuid;oleaut32;shlwapi;secur32;m;psapi;advapi32;shell32;avformat;avicap32;gdi32;psapi;ole32;strmiids;uuid;oleaut32;shlwapi;secur32;m;psapi;advapi32;shell32;avcodec;avicap32;gdi32;psapi;ole32;strmiids;uuid;oleaut32;shlwapi;secur32;m;psapi;advapi32;shell32;avutil;m
-- LIBAVDEVICE found
-- Checking for one of the modules 'libavformat'
-- LIBAVFORMAT LIBRARY_DIRS: /workspace/i686/dep-cache/libffmpeg/lib
-- LIBAVFORMAT INCLUDE_DIRS: /workspace/i686/dep-cache/libffmpeg/include
-- LIBAVFORMAT CFLAGS_OTHER: 
-- LIBAVFORMAT LIBRARIES:    avformat;avicap32;gdi32;psapi;ole32;strmiids;uuid;oleaut32;shlwapi;secur32;m;psapi;advapi32;shell32;avcodec;avicap32;gdi32;psapi;ole32;strmiids;uuid;oleaut32;shlwapi;secur32;m;psapi;advapi32;shell32;avutil;m
-- LIBAVFORMAT found
-- Checking for one of the modules 'libavutil'
-- LIBAVUTIL LIBRARY_DIRS: /workspace/i686/dep-cache/libffmpeg/lib
-- LIBAVUTIL INCLUDE_DIRS: /workspace/i686/dep-cache/libffmpeg/include
-- LIBAVUTIL CFLAGS_OTHER: 
-- LIBAVUTIL LIBRARIES:    avutil;m
-- LIBAVUTIL found
-- Checking for one of the modules 'libexif'
-- LIBEXIF LIBRARY_DIRS: /workspace/i686/dep-cache/libexif/lib
-- LIBEXIF INCLUDE_DIRS: /workspace/i686/dep-cache/libexif/include
-- LIBEXIF CFLAGS_OTHER: 
-- LIBEXIF LIBRARIES:    exif
-- LIBEXIF found
-- Checking for one of the modules 'libqrencode'
-- LIBQRENCODE LIBRARY_DIRS: /workspace/i686/dep-cache/libqrencode/lib
-- LIBQRENCODE INCLUDE_DIRS: /workspace/i686/dep-cache/libqrencode/include
-- LIBQRENCODE CFLAGS_OTHER: 
-- LIBQRENCODE LIBRARIES:    qrencode
-- LIBQRENCODE found
-- Checking for one of the modules 'libsodium'
-- LIBSODIUM LIBRARY_DIRS: /workspace/i686/dep-cache/libsodium/lib
-- LIBSODIUM INCLUDE_DIRS: /workspace/i686/dep-cache/libsodium/include
-- LIBSODIUM CFLAGS_OTHER: 
-- LIBSODIUM LIBRARIES:    sodium
-- LIBSODIUM found
-- Checking for one of the modules 'libswscale'
-- LIBSWSCALE LIBRARY_DIRS: /workspace/i686/dep-cache/libffmpeg/lib
-- LIBSWSCALE INCLUDE_DIRS: /workspace/i686/dep-cache/libffmpeg/include
-- LIBSWSCALE CFLAGS_OTHER: 
-- LIBSWSCALE LIBRARIES:    swscale;m;avutil;m
-- LIBSWSCALE found
-- Checking for one of the modules 'sqlcipher'
-- SQLCIPHER LIBRARY_DIRS: /workspace/i686/dep-cache/libsqlcipher/lib
-- SQLCIPHER INCLUDE_DIRS: /workspace/i686/dep-cache/libsqlcipher/include/sqlcipher
-- SQLCIPHER CFLAGS_OTHER: 
-- SQLCIPHER LIBRARIES:    sqlcipher
-- SQLCIPHER found
-- Checking for one of the modules 'vpx'
-- VPX LIBRARY_DIRS: /workspace/i686/dep-cache/libvpx/lib
-- VPX INCLUDE_DIRS: /workspace/i686/dep-cache/libvpx/include
-- VPX CFLAGS_OTHER: 
-- VPX LIBRARIES:    vpx;m
-- VPX found
-- Checking for one of the modules 'toxcore'
-- TOXCORE LIBRARY_DIRS: /workspace/i686/dep-cache/libsodium/lib;/workspace/i686/dep-cache/libtoxcore/lib
-- TOXCORE INCLUDE_DIRS: /workspace/i686/dep-cache/libsodium/include;/workspace/i686/dep-cache/libtoxcore/include
-- TOXCORE CFLAGS_OTHER: 
-- TOXCORE LIBRARIES:    toxcore;pthread;ws2_32;iphlpapi;toxcrypto;toxnetwork;toxdht;toxnetcrypto;toxfriends;toxmessenger;toxgroup;sodium
-- TOXCORE found
-- Checking for one of the modules 'toxav'
-- TOXAV LIBRARY_DIRS: /workspace/i686/dep-cache/libopus/lib;/workspace/i686/dep-cache/libsodium/lib;/workspace/i686/dep-cache/libtoxcore/lib;/workspace/i686/dep-cache/libvpx/lib
-- TOXAV INCLUDE_DIRS: /workspace/i686/dep-cache/libopus/include/opus;/workspace/i686/dep-cache/libsodium/include;/workspace/i686/dep-cache/libtoxcore/include;/workspace/i686/dep-cache/libvpx/include
-- TOXAV CFLAGS_OTHER: 
-- TOXAV LIBRARIES:    toxav;opus;toxcore;pthread;ws2_32;iphlpapi;toxcrypto;toxnetwork;toxdht;toxnetcrypto;toxfriends;toxmessenger;toxgroup;sodium;vpx;m
-- TOXAV found
-- Checking for one of the modules 'toxencryptsave'
-- TOXENCRYPTSAVE LIBRARY_DIRS: /workspace/i686/dep-cache/libsodium/lib;/workspace/i686/dep-cache/libtoxcore/lib
-- TOXENCRYPTSAVE INCLUDE_DIRS: /workspace/i686/dep-cache/libsodium/include;/workspace/i686/dep-cache/libtoxcore/include
-- TOXENCRYPTSAVE CFLAGS_OTHER: 
-- TOXENCRYPTSAVE LIBRARIES:    toxencryptsave;toxcore;pthread;ws2_32;iphlpapi;toxcrypto;toxnetwork;toxdht;toxnetcrypto;toxfriends;toxmessenger;toxgroup;sodium
-- TOXENCRYPTSAVE found
-- Checking for one of the modules 'openal'
-- OPENAL LIBRARY_DIRS: /workspace/i686/dep-cache/libopenal/lib
-- OPENAL INCLUDE_DIRS: /workspace/i686/dep-cache/libopenal/include;/workspace/i686/dep-cache/libopenal/include/AL
-- OPENAL CFLAGS_OTHER: 
-- OPENAL LIBRARIES:    OpenAL32
-- OPENAL found
-- Checking for one of the modules 'openssl'
-- OPENSSL LIBRARY_DIRS: /workspace/i686/dep-cache/libopenssl/lib
-- OPENSSL INCLUDE_DIRS: /workspace/i686/dep-cache/libopenssl/include
-- OPENSSL CFLAGS_OTHER: 
-- OPENSSL LIBRARIES:    ssl;crypto
-- OPENSSL found
CMake Warning at cmake/Dependencies.cmake:205 (message):
  Not using platform extensions, dependencies not found
Call Stack (most recent call first):
  CMakeLists.txt:80 (include)


-- FILTERAUDIO LIBRARY_DIRS: 
-- FILTERAUDIO INCLUDE_DIRS: /workspace/i686/dep-cache/libfilteraudio/include
-- FILTERAUDIO CFLAGS_OTHER: 
-- FILTERAUDIO LIBRARIES:    /workspace/i686/dep-cache/libfilteraudio/lib/libfilteraudio.a
-- FILTERAUDIO found
-- using filteraudio
-- CMAKE_C_FLAGS: 
-- CMAKE_CXX_FLAGS:  -std=c++11 -fno-exceptions -fno-rtti -fPIE -fPIC -Wstrict-overflow -Wstrict-aliasing -Werror
-- Configuring done
-- Generating done
-- Build files have been written to: /build/qtox/build
+ make
Scanning dependencies of target test_toxid_automoc
[  1%] Automatic moc for target test_toxid
Generating toxid_test.moc
[  1%] Built target test_toxid_automoc
Scanning dependencies of target qtox_static_automoc
[  1%] Automatic moc for target qtox_static
Generating moc_audio.cpp
Generating moc_openal.cpp
Generating moc_openal2.cpp
Generating moc_chatlinecontent.cpp
Generating moc_chatlinecontentproxy.cpp
Generating moc_chatlog.cpp
Generating moc_filetransferwidget.cpp
Generating moc_notificationicon.cpp
Generating moc_spinner.cpp
Generating moc_text.cpp
Generating moc_timestamp.cpp
Generating moc_customtextdocument.cpp
Generating moc_core.cpp
Generating moc_coreav.cpp
Generating moc_ipc.cpp
Generating moc_aboutfriend.cpp
Generating moc_iaboutfriend.cpp
Generating moc_contact.cpp
Generating moc_friend.cpp
Generating moc_group.cpp
Generating moc_toxuri.cpp
Generating moc_nexus.cpp
Generating moc_rawdatabase.cpp
Generating moc_offlinemsgengine.cpp
Generating moc_profile.cpp
Generating moc_settings.cpp
Generating moc_smileypack.cpp
Generating moc_camerasource.cpp
Generating moc_corevideosource.cpp
Generating moc_genericnetcamview.cpp
Generating moc_netcamview.cpp
Generating moc_videosource.cpp
Generating moc_videosurface.cpp
Generating moc_aboutfriendform.cpp
Generating moc_categorywidget.cpp
Generating moc_circlewidget.cpp
Generating moc_contentdialog.cpp
Generating moc_emoticonswidget.cpp
Generating moc_addfriendform.cpp
Generating moc_chatform.cpp
Generating moc_filesform.cpp
Generating moc_genericchatform.cpp
Generating moc_groupchatform.cpp
Generating moc_groupinviteform.cpp
Generating moc_groupinvitewidget.cpp
Generating moc_loadhistorydialog.cpp
Generating moc_profileform.cpp
Generating moc_setpassworddialog.cpp
Generating moc_aboutform.cpp
Generating moc_advancedform.cpp
Generating moc_avform.cpp
Generating moc_generalform.cpp
Generating moc_genericsettings.cpp
Generating moc_privacyform.cpp
Generating moc_userinterfaceform.cpp
Generating moc_verticalonlyscroller.cpp
Generating moc_settingswidget.cpp
Generating moc_tabcompleter.cpp
Generating moc_friendlistlayout.cpp
Generating moc_friendlistwidget.cpp
Generating moc_friendwidget.cpp
Generating moc_genericchatitemwidget.cpp
Generating moc_genericchatroomwidget.cpp
Generating moc_groupwidget.cpp
Generating moc_gui.cpp
Generating moc_loginscreen.cpp
Generating moc_maskablepixmapwidget.cpp
Generating moc_notificationedgewidget.cpp
Generating moc_passwordedit.cpp
Generating moc_qrwidget.cpp
Generating moc_systemtrayicon.cpp
Generating moc_activatedialog.cpp
Generating moc_adjustingscrollarea.cpp
Generating moc_callconfirmwidget.cpp
Generating moc_chattextedit.cpp
Generating moc_croppinglabel.cpp
Generating moc_flyoutoverlaywidget.cpp
Generating moc_friendrequestdialog.cpp
Generating moc_profileimporter.cpp
Generating moc_removefrienddialog.cpp
Generating moc_screengrabberchooserrectitem.cpp
Generating moc_screengrabberoverlayitem.cpp
Generating moc_screenshotgrabber.cpp
Generating moc_toolboxgraphicsitem.cpp
Generating moc_widget.cpp
[  1%] Built target qtox_static_automoc
[  1%] Generating zh_TW.qm
Updating '/build/qtox/build/zh_TW.qm'...
    Generated 173 translation(s) (124 finished and 49 unfinished)
    Ignored 465 untranslated source text(s)
[  1%] Generating ui_filetransferwidget.h
[  1%] Generating ui_loginscreen.h
[  2%] Generating ui_mainwindow.h
[  2%] Generating ui_aboutfriendform.h
[  3%] Generating ui_loadhistorydialog.h
[  3%] Generating ui_profileform.h
[  3%] Generating ui_removefrienddialog.h
[  4%] Generating ui_setpassworddialog.h
[  4%] Generating ui_aboutsettings.h
[  5%] Generating ui_advancedsettings.h
[  5%] Generating ui_avform.h
[  6%] Generating ui_generalsettings.h
[  6%] Generating ui_privacysettings.h
[  6%] Generating ui_userinterfacesettings.h
[  7%] Generating ar.qm
Updating '/build/qtox/build/ar.qm'...
    Generated 606 translation(s) (575 finished and 31 unfinished)
    Ignored 32 untranslated source text(s)
[  7%] Generating be.qm
Updating '/build/qtox/build/be.qm'...
    Generated 620 translation(s) (593 finished and 27 unfinished)
    Ignored 18 untranslated source text(s)
[  8%] Generating bg.qm
Updating '/build/qtox/build/bg.qm'...
    Generated 610 translation(s) (573 finished and 37 unfinished)
    Ignored 28 untranslated source text(s)
[  8%] Generating cs.qm
Updating '/build/qtox/build/cs.qm'...
    Generated 524 translation(s) (463 finished and 61 unfinished)
    Ignored 114 untranslated source text(s)
[  8%] Generating da.qm
Updating '/build/qtox/build/da.qm'...
    Generated 107 translation(s) (51 finished and 56 unfinished)
    Ignored 531 untranslated source text(s)
[  9%] Generating de.qm
Updating '/build/qtox/build/de.qm'...
    Generated 617 translation(s) (587 finished and 30 unfinished)
    Ignored 21 untranslated source text(s)
[  9%] Generating el.qm
Updating '/build/qtox/build/el.qm'...
    Generated 616 translation(s) (589 finished and 27 unfinished)
    Ignored 22 untranslated source text(s)
[ 10%] Generating eo.qm
Updating '/build/qtox/build/eo.qm'...
    Generated 269 translation(s) (247 finished and 22 unfinished)
    Ignored 369 untranslated source text(s)
[ 10%] Generating es.qm
Updating '/build/qtox/build/es.qm'...
    Generated 587 translation(s) (534 finished and 53 unfinished)
    Ignored 51 untranslated source text(s)
[ 11%] Generating et.qm
Updating '/build/qtox/build/et.qm'...
    Generated 620 translation(s) (593 finished and 27 unfinished)
    Ignored 18 untranslated source text(s)
[ 11%] Generating fi.qm
Updating '/build/qtox/build/fi.qm'...
    Generated 366 translation(s) (347 finished and 19 unfinished)
    Ignored 272 untranslated source text(s)
[ 11%] Generating fr.qm
Updating '/build/qtox/build/fr.qm'...
    Generated 621 translation(s) (595 finished and 26 unfinished)
    Ignored 17 untranslated source text(s)
[ 12%] Generating he.qm
Updating '/build/qtox/build/he.qm'...
    Generated 14 translation(s) (3 finished and 11 unfinished)
    Ignored 624 untranslated source text(s)
[ 12%] Generating hr.qm
Updating '/build/qtox/build/hr.qm'...
    Generated 251 translation(s) (156 finished and 95 unfinished)
    Ignored 387 untranslated source text(s)
[ 13%] Generating hu.qm
Updating '/build/qtox/build/hu.qm'...
    Generated 600 translation(s) (554 finished and 46 unfinished)
    Ignored 38 untranslated source text(s)
[ 13%] Generating it.qm
Updating '/build/qtox/build/it.qm'...
    Generated 606 translation(s) (568 finished and 38 unfinished)
    Ignored 32 untranslated source text(s)
[ 14%] Generating ja.qm
Updating '/build/qtox/build/ja.qm'...
    Generated 477 translation(s) (16 finished and 461 unfinished)
    Ignored 161 untranslated source text(s)
[ 14%] Generating jbo.qm
Updating '/build/qtox/build/jbo.qm'...
    Generated 12 translation(s) (7 finished and 5 unfinished)
    Ignored 626 untranslated source text(s)
[ 14%] Generating ko.qm
Updating '/build/qtox/build/ko.qm'...
    Generated 201 translation(s) (145 finished and 56 unfinished)
    Ignored 437 untranslated source text(s)
[ 15%] Generating lt.qm
Updating '/build/qtox/build/lt.qm'...
    Generated 620 translation(s) (593 finished and 27 unfinished)
    Ignored 18 untranslated source text(s)
[ 15%] Generating nl.qm
Updating '/build/qtox/build/nl.qm'...
    Generated 421 translation(s) (317 finished and 104 unfinished)
    Ignored 217 untranslated source text(s)
[ 16%] Generating no_nb.qm
Updating '/build/qtox/build/no_nb.qm'...
    Generated 620 translation(s) (569 finished and 51 unfinished)
    Ignored 18 untranslated source text(s)
[ 16%] Generating pl.qm
Updating '/build/qtox/build/pl.qm'...
    Generated 605 translation(s) (553 finished and 52 unfinished)
    Ignored 33 untranslated source text(s)
[ 16%] Generating pr.qm
Updating '/build/qtox/build/pr.qm'...
    Generated 238 translation(s) (177 finished and 61 unfinished)
    Ignored 400 untranslated source text(s)
[ 17%] Generating pt.qm
Updating '/build/qtox/build/pt.qm'...
    Generated 446 translation(s) (334 finished and 112 unfinished)
    Ignored 192 untranslated source text(s)
[ 17%] Generating ro.qm
Updating '/build/qtox/build/ro.qm'...
    Generated 621 translation(s) (593 finished and 28 unfinished)
    Ignored 17 untranslated source text(s)
[ 18%] Generating ru.qm
Updating '/build/qtox/build/ru.qm'...
    Generated 621 translation(s) (593 finished and 28 unfinished)
    Ignored 17 untranslated source text(s)
[ 18%] Generating sk.qm
Updating '/build/qtox/build/sk.qm'...
    Generated 563 translation(s) (473 finished and 90 unfinished)
    Ignored 75 untranslated source text(s)
[ 19%] Generating sl.qm
Updating '/build/qtox/build/sl.qm'...
    Generated 246 translation(s) (147 finished and 99 unfinished)
    Ignored 392 untranslated source text(s)
[ 19%] Generating sv.qm
Updating '/build/qtox/build/sv.qm'...
    Generated 621 translation(s) (593 finished and 28 unfinished)
    Ignored 17 untranslated source text(s)
[ 19%] Generating sw.qm
Updating '/build/qtox/build/sw.qm'...
    Generated 0 translation(s) (0 finished and 0 unfinished)
    Ignored 638 untranslated source text(s)
[ 20%] Generating ta.qm
Updating '/build/qtox/build/ta.qm'...
    Generated 338 translation(s) (222 finished and 116 unfinished)
    Ignored 300 untranslated source text(s)
[ 20%] Generating tr.qm
Updating '/build/qtox/build/tr.qm'...
    Generated 410 translation(s) (376 finished and 34 unfinished)
    Ignored 228 untranslated source text(s)
[ 21%] Generating ug.qm
Updating '/build/qtox/build/ug.qm'...
    Generated 46 translation(s) (12 finished and 34 unfinished)
    Ignored 592 untranslated source text(s)
[ 21%] Generating uk.qm
Updating '/build/qtox/build/uk.qm'...
    Generated 598 translation(s) (468 finished and 130 unfinished)
    Ignored 40 untranslated source text(s)
[ 22%] Generating zh_CN.qm
Updating '/build/qtox/build/zh_CN.qm'...
    Generated 619 translation(s) (593 finished and 26 unfinished)
    Ignored 19 untranslated source text(s)
Scanning dependencies of target qtox_static
[ 22%] Building CXX object CMakeFiles/qtox_static.dir/src/audio/audio.cpp.obj
/build/qtox/src/audio/audio.cpp:1:0: error: -fPIC ignored for target (all code is position independent) [-Werror]
 /*
 ^
cc1plus: all warnings being treated as errors
CMakeFiles/qtox_static.dir/build.make:267: recipe for target 'CMakeFiles/qtox_static.dir/src/audio/audio.cpp.obj' failed
make[2]: *** [CMakeFiles/qtox_static.dir/src/audio/audio.cpp.obj] Error 1
make[1]: *** [CMakeFiles/qtox_static.dir/all] Error 2
CMakeFiles/Makefile2:181: recipe for target 'CMakeFiles/qtox_static.dir/all' failed
make: *** [all] Error 2
Makefile:138: recipe for target 'all' failed
```

**Comment by zetok** [comment]
Rather than removing `-Werror`, wouldn't removing `-fPIC` from args for windows fix it?

**Comment by zetok** [comment]
@nurupo also, background: #4272

**Comment by sudden6** [comment]
I don't think disabling `-Werror` is the best solution, most of the time there's a reason for the errors and if there's really no reason, we can specifically disable that offending one.

**Comment by nurupo** [comment]
>most of the time there's a reason for the errors 

Those are warnings, `-Werror` turns all warnings into errors, meaning that if you get at least 1 warning, qTox compilation would halt.

>we can specifically disable that offending one.

You can, but what about qTox users who want to compile qTox but can't because their compiler spits a warning?

Let me repeat myself for the fourth time.

1. `-Werror` is not future proof. You might get no warnings on gcc 4.9, but a future version of gcc might add new warnings, which would turn into errors by `-Werror` and not let users using those newer compilers build qTox.
2. `-Werror` is not compiler proof. Different compilers might generate different sets of warnings. If you get no warnings using compiler X, you might get them using compiler Y. People who try to compile qTox using compiler Y won't be able to do so.
3.  `-Werror` is not platform proof. You might get new sets of warnings when you try to compile for Windows, FreeBSD, etc., which again would not let users to get qTox to build.

What is the reason for having `-Werror` in the first place? If it's to reduce number of compiler warnings, then enabling `-Werror` only in a debug build, or having a separate cmake option for enabling it like c-toxcore does, and using those in CI is enough. No need to break builds for users, only for developers.

**Comment by sudden6** [comment]
Link to the list of failing builds: https://build.tox.chat/job/qtox-cmake-build-script_src/ due to this problem.

**Comment by Diadlo** [comment]
Fixed in #4712
